syntax = "proto3";

package dds;

// MAP PROTO API
message MapKey {
    string map_key = 1;
}

message MapValue {
    int32 map_value = 1;
}

message KeyValue {
    string map_key = 1;
    int32 map_value = 2;
}

message Void {}

message Int {
    int32 num = 1;
}

message Bool {
    bool flag = 1;
}

service MapService {
    rpc GetValue (MapKey) returns (MapValue) {}
    rpc SetValue (KeyValue) returns (Void) {}
    rpc Size (Void) returns (Int) {}
    rpc Empty (Void) returns (Bool) {}
    rpc Remove (MapKey) returns (Void) {}
    rpc Clear (Void) returns (Void) {}
}

// Consul healthcheck
message HealthCheckRequest {}

message HealthCheckResponse {}

service HealthCheck {
    rpc Check(HealthCheckRequest) returns (HealthCheckResponse) {}
}
